lib/src/standalone.dart

```diff
@@ -299,20 +266,5 @@ Future<List<int>> _dartExecutable(String os, String arch) async {
       .content as List<int>;
 }
 
-/// Throws an error if [os] and [arch] aren't a valid combination.
-///
-/// This is just intended to guard against programmer error within `cli_pkg`.
-void _verifyOsAndArch(String os, String arch) {
-  if (!["macos", "windows", "linux"].contains(os)) {
-    fail("Unknown operating system $os!");
-  } else if (!["ia32", "x64", "arm64"].contains(arch)) {
-    fail("Unknown architecture $arch!");
-  } else if (os == "macos" && arch == "ia32") {
-    fail("Dart doesn't support 32-bit Mac OS!");
-  } else if (os == "windows" && arch == "arm64") {
-    fail("Dart doesn't support Windows on ARM!");
-  }
-}
-
```

Comment:
I think this is still worth including. It's a helpful safeguard in case we do future refactors (although simplifying it using the os/architecture map could work).